packages/rn-tester/js/RNTesterAppShared.js

```diff
@@ -0,0 +1,244 @@
+
+  return (
+    <DisplayIfVisible isVisible={isVisible}>
+      <Header title={title} theme={theme} />
+      <DisplayIfVisible isVisible={screen === Screens.COMPONENTS}>
+        <RNTesterExampleList
+          sections={examplesList.components}
+          toggleBookmark={toggleBookmark}
+          handleExampleCardPress={handleExampleCardPress}
+        />
+      </DisplayIfVisible>
+      <DisplayIfVisible isVisible={screen === Screens.APIS}>
+        <RNTesterExampleList
+          sections={examplesList.apis}
+          toggleBookmark={toggleBookmark}
+          handleExampleCardPress={handleExampleCardPress}
+        />
+      </DisplayIfVisible>
+      <DisplayIfVisible isVisible={screen === Screens.BOOKMARKS}>
+        {isBookmarkEmpty ? (
+          <RNTesterEmptyBookmarksState />
+        ) : (
+          <RNTesterExampleList
+            sections={examplesList.bookmarks}
+            toggleBookmark={toggleBookmark}
+            handleExampleCardPress={handleExampleCardPress}
+          />
+        )}
+      </DisplayIfVisible>
+    </DisplayIfVisible>
+  );
+};
+
+const RNTesterApp = (): React.Node => {
+  const [state, dispatch] = useAsyncStorageReducer(
+    RNTesterReducer,
+    initialState,
+    APP_STATE_KEY,
+  );
+
+  React.useEffect(() => {
+    getInitialStateFromAsyncStorage(APP_STATE_KEY).then(
+      initialStateFromStorage => {
+        dispatch({
+          type: RNTesterActionsType.INIT_FROM_STORAGE,
+          data: initialStateFromStorage,
+        });
+      },
+    );
+  }, [dispatch]);
+
+  const {openExample, screen, bookmarks, recentlyUsed} = state;
+
+  const examplesList = React.useMemo(
+    () =>
+      getExamplesListWithBookmarksAndRecentlyUsed({bookmarks, recentlyUsed}),
+    [bookmarks, recentlyUsed],
+  );
+
+  const handleBackPress = React.useCallback(() => {
+    if (openExample) {
+      dispatch({type: RNTesterActionsType.BACK_BUTTON_PRESS});
+    }
+  }, [dispatch, openExample]);
+
+  // Setup hardware back button press listener
+  React.useEffect(() => {
+    BackHandler.addEventListener('hardwareBackPress', () => {
+      if (openExample) {
+        handleBackPress();
+        return true;
+      }
+      return false;
+    });
+  }, [openExample, handleBackPress]);
+
+  const handleExampleCardPress = React.useCallback(
+    ({exampleType, key}) => {
+      dispatch({
+        type: RNTesterActionsType.EXAMPLE_CARD_PRESS,
+        data: {exampleType, key},
+      });
+    },
+    [dispatch],
+  );
+
+  const toggleBookmark = React.useCallback(
+    ({exampleType, key}) => {
+      dispatch({
+        type: RNTesterActionsType.BOOKMARK_PRESS,
+        data: {exampleType, key},
+      });
+    },
+    [dispatch],
+  );
+
+  const handleNavBarPress = React.useCallback(
+    args => {
+      dispatch({
+        type: RNTesterActionsType.NAVBAR_PRESS,
+        data: {screen: args.screen},
+      });
+    },
+    [dispatch],
+  );
+
+  const colorScheme = useColorScheme();
```

Comment:
Same here, would move all the hooks you use up to the top of the function.

Reply:
Done!